### PR TITLE
Introduce tz address

### PR DESCRIPTION
**Pull request by mspwn**
Just to make the implementation of `transfer` possible.

### CODE REVIEW

File: lib/yourbones/address.ml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    else if String.starts_with ~prefix:"tz3" str
+    then Ok (`P256, p256_prefix)
+    else if String.starts_with ~prefix:"tz4" str
+    then Ok (`Bls12_381, bls12_381_prefix)
+    else if String.starts_with ~prefix:"KT1" str
+    then Ok (`NoCurve, nocurve_prefix)
+    else if String.starts_with ~prefix:"sr1" str
+    then Ok (`SR1, sr1_prefix)
+    else Error (`Address_invalid_prefix str)
+  with
```
Comment (xhtmlboi) [inline lib/yourbones/address.ml]: Each prefix starts with three characters, so I propose to take a slightly more naive approach (not-tested pseudocode): 
 
```ocaml
let from_string str = 
  if String.length str <> 36 then 
     Error (`Address_invalid_length str)
  else match (str.[0], str.[1], str.[2) with
   | ('t', 'z', '1') -> ... 
```

File: lib/yourbones/address.ml [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  | `SR1
+  ]
+
+type tz1 = [ `Ed25519 ]
+type tz2 = [ `Secp256k1 ]
+type tz3 = [ `P256 ]
+type tz4 = [ `Bls12_381 ]
+type kt1 = [ `NoCurve ]
+type sr1 = [ `SR1 ]
+type 'a t = 'a * string constraint 'a = kind
```
Comment (xhtmlboi) [inline lib/yourbones/address.ml]: I find it very hard to see the point of the phantom type here. 
I think that initially, we can imagine something very simple, like : 
```ocaml
type t = private 
  | Tz1 of string
  | Tz2 of string 
  | Tz3 of string
  | Kt1 of string (* maybe *)
```
Or separate the notion of implicit account vs. originated account.
I don't have a precise idea of a useful and conveinant design but I think that supporting Tz4 and Sr1 and indexing the keys by their cryptographic approach is not necessary at this time.

**Review by xhtmlboi — CHANGES_REQUESTED** [review]
I've put forward a few ideas, but as the most useful pair of functions will be `(from|to)_string`, I think we'll very quickly lose interest in the phantom type. So I propose to simplify the design.

**Review by xhtmlboi — APPROVED** [review]
That seems sufficient for an initial exploratory phase. I'll let @xvw approve and merge!

**Review by xvw — APPROVED** [review]
\o/

**Pull request merged**